### PR TITLE
[OTEL-2304] export obfuscateStatsGroup function

### PR DESCRIPTION
**Pull request by jackgopack4**
### What does this PR do?
removes `*Agent` type receiver from obfuscateStatsGroup function and exports for use in other Datadog components

### Motivation
enable obfuscation for APM trace metrics for OTel Collector customers inside datadogconnector: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35401

We don't want to have to duplicate this function or duplicate instantiating an entire Agent inside datadogconnector component; if this is accepted and merged I can just import this function and pass in the obfuscator we instantiate there.

### Describe how you validated your changes

made change to existing unit test

### Possible Drawbacks / Trade-offs
need to pass obfuscator pointer to the function instead of just calling on Agent struct pointer; any downstream dependencies that import this function

### Additional Notes
If we know that any other datadog or downstream dependencies are using this function currently; I'd be happy to break out a helper function that takes obfuscator as a pointer and keep the existing signature as a shell for compatibility. Let me know if this is the case

### CODE REVIEW

File: pkg/trace/agent/obfuscate.go [hunk 1]
```diff
@@ -110,9 +110,7 @@ func (a *Agent) obfuscateSpan(span *pb.Span) {
 	}
 }
 
-func (a *Agent) obfuscateStatsGroup(b *pb.ClientGroupedStats) {
-	o := a.lazyInitObfuscator()
-
+func obfuscateStatsGroup(o *obfuscate.Obfuscator, b *pb.ClientGroupedStats) {
```
Comment (jackgopack4) [inline pkg/trace/agent/obfuscate.go:113]: if we have other dependencies not in Agent using this function; I could break out this change to a helper function instead of changing the signature

**Review by songy23 — APPROVED** [review]

File: pkg/trace/agent/agent.go [hunk 2]
```diff
@@ -535,7 +535,8 @@ func (a *Agent) processStats(in *pb.ClientStatsPayload, lang, tracerVersion, con
 			if !a.Blacklister.AllowsStat(b) {
 				continue
 			}
-			a.obfuscateStatsGroup(b)
+			obfuscator := a.lazyInitObfuscator()
```
Comment (ichinaski) [inline pkg/trace/agent/agent.go:538]: @ajgajg1134 is there any reason why this initialization happens outside the constructor? I think it could simplify the code a bit if we move this to the Agent constructor, especially now that the obfuscation function will be exported?

**Review by ichinaski — COMMENTED** [review]
LGTM! Just a small comment around the lazy initialization of the Obfuscator, I wonder if we can further simplify this part

File: pkg/trace/agent/obfuscate.go [hunk 3]
```diff
@@ -110,7 +110,8 @@ func (a *Agent) obfuscateSpan(span *pb.Span) {
 	}
 }
 
-func obfuscateStatsGroup(o *obfuscate.Obfuscator, b *pb.ClientGroupedStats) {
+// ObfuscateStatsGroup obfuscates sensitive data to enable APM Trace Metrics calculation
```
Comment (songy23) [inline pkg/trace/agent/obfuscate.go:113]: This comment is not quite accurate, I would rephrase as something like
```
// ObfuscateStatsGroup obfuscates sensitive data in APM stats payloads
```

File: pkg/trace/agent/agent.go (hunk 2 above)
Comment (ajgajg1134) [reply pkg/trace/agent/agent.go:538]: Huh, yeah it looks like it used to be initialized in the newAgent but very recently was moved here: https://github.com/DataDog/datadog-agent/pull/31336 @sethsamuel do you happen to recall why you moved the obfuscator initialization to be lazy init outside the NewAgent func?

I'm not entirely sure _why_ it was moved but it seems like it should probably be moved back to the agent so that a "NewAgent" can know that it safely can use the obfuscator field and doesn't have to always remember to lazy init it for every use :P

Comment (jackgopack4) [reply pkg/trace/agent/agent.go:538]: seems reasonable to me; we are going to be calling `obfuscate.NewObfuscator` in the datadog connector

Comment (jackgopack4) [reply pkg/trace/agent/agent.go:538]: looks like @sethsamuel is OOO until Monday

Comment (jackgopack4) [reply pkg/trace/agent/agent.go:538]: this may not be the exact function we need to use for datadogconnector; I think it may be missing out on more data types. We are calculating more than just sql, cassandra, and redis metrics in connector so we will either need to enhance this function, or write our own. Converting this PR to draft for now, especially since I'm out for two weeks and will be calking with APM Astra after the holidays @jdgumz

**Comment by jackgopack4** [comment]
closed in favor of #32829 and https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37457
